deps/src/main/java/com/microsoft/azure/sdk/iot/deps/convention/PayloadSerializer.java

```
@@ -0,0 +1,85 @@
+     * </p>
+     * @return A string representing the content type to use when sending a payload.
+     */
+    @Getter
+    String ContentType;
+
+    /**
+     * Serialize the specified object to a string.
+     * @param objectToSerialize Object to serialize.
+     * @return A serialized string of the object.
+     */
+    public abstract String serializeToString(Object objectToSerialize);
+
+    /**
+     * Convert the serialized string to an object.
+     * @param stringToDeserialize String to deserialize.
+     * @param <T> The type you want to return.
+     * @return A fully deserialized type.
+     */
+    public abstract <T> T deserializeToType(String stringToDeserialize);
+
+    /**
+     * Converts the object using the serializer.
+     *
+     * <p>
+     *     This class is used by the PayloadCollection-based classes to attempt to convert from the native serializer type to the desired type.
+     *     When you implement this you need to be aware of what type your serializer will use for anonymous types.
+     * </p>
+     * @param objectToConvert The object to convert.
+     * @param <T> The type to convert to.
+     * @return A converted object.
+     */
+    public abstract <T> T convertFromObject(Object objectToConvert);
+
+    /**
+     * Gets a nested property from the serialized data.
+     * <p>
+     *     This is used internally by our PayloadCollection-based classes to attempt to get a property of the underlying object.
+     *     An example of this would be a property under the component.
+     * </p>
+     * @param nestedObject The object that might contain the nested property.
+     * @param propertyName The name of the property to be retrieved.
+     * @param <T> The type to convert the retrieved property to.
+     * @return {@code true} if the nested object contains an element with the specified key; otherwise, it returns {@code false}.
+     */
+    public abstract <T> T getNestedObjectValue(Object nestedObject, String propertyName);
```

since java doesn't follow the TryGet pattern strictly we'll need to declare and throw an exception for these cases-> key not found, value could not be cast to T

---

Will defer to Tim on this but in most cases Java returns null when something is not found instead of throwing an exception